theano/compile/builders.py

```diff
@@ -133,6 +135,59 @@ def perform(self, node, inputs, outputs):
             # we wont need this copy anymore
             output[0] = variable.copy()
 
+    def connection_pattern(self, node):
```

Comment:
Sorry I mixed  two PR.  Buy they are quit the same.
I will have a look and see if i can merge two codes. Can you provide some suggestion for test case?